test/s3select_test.cpp

```diff
@@ -280,25 +280,25 @@ TEST(TestS3SElect, floatnan_compare_operator)
   ASSERT_EQ( d >= c10, false );
   ASSERT_EQ( d < a10, false );
   ASSERT_EQ( d <= b11, false );
-  ASSERT_EQ( d != a10, true );
-  ASSERT_EQ( d != e, true );
-  ASSERT_EQ( d == a10, false );
+  ASSERT_EQ( d != a10, false );
+  ASSERT_EQ( d != e, false );
+ // ASSERT_EQ( d == a10, false );
 }
 
 TEST(TestS3SElect, null_arithmetic_operator)
 {
   value a(7), d, e(0);
   d.setnull();
-  ASSERT_EQ((a + d).to_string(), "nan" );
-  ASSERT_EQ((a - d).to_string(), "nan" );
-  ASSERT_EQ((a * d).to_string(), "nan" );
-  ASSERT_EQ((a / d).to_string(), "nan" ); 
-  ASSERT_EQ((a / e).to_string(), "nan" ); 
-  ASSERT_EQ((d + a).to_string(), "nan" );
-  ASSERT_EQ((d - a).to_string(), "nan" );
-  ASSERT_EQ((d * a).to_string(), "nan" );
-  ASSERT_EQ((d / a).to_string(), "nan" ); 
-  ASSERT_EQ((e / a).to_string(), "nan" );
+  ASSERT_EQ((a + d).to_string(), "null" );
+  ASSERT_EQ((a - d).to_string(), "null" );
+  ASSERT_EQ((a * d).to_string(), "null" );
+  ASSERT_EQ((a / d).to_string(), "null" ); 
+  ASSERT_EQ((a / e).to_string(), "null" ); 
```

Comment:
divide number by zero should raise exception

Reply:
Isn't null/0 null?

Reply:
yes
null/0 = null